### PR TITLE
fix(es/minifier): Make name mangler respect `toplevel`

### PR DESCRIPTION
**Pull request by kdy1**
**Description:**

**Related issue:**

 - Closes https://github.com/swc-project/swc/issues/6418.

### CODE REVIEW

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_ecma_minifier --breaking

File: crates/swc_ecma_transforms_optimization/src/simplify/dce/mod.rs [hunk 1]
```diff
@@ -633,6 +633,14 @@ impl TreeShaker {
             } else if !self.in_block_stmt {
                 return false;
             }
+
+            // Abort if the variable is declared on top level scope.
+            let ix = self.data.graph_ix.get_index_of(&name);
```
Comment (kdy1) [inline crates/swc_ecma_transforms_optimization/src/simplify/dce/mod.rs:638]: This is for https://github.com/swc-project/swc/issues/4386

Comment (kdy1) [reply crates/swc_ecma_transforms_optimization/src/simplify/dce/mod.rs:638]: I didn't reverted this as this is correct change anyway

Comment (kdy1) [reply crates/swc_ecma_transforms_optimization/src/simplify/dce/mod.rs:638]: But I'll extract this into another PR.

**Review by jridgewell — APPROVED** [review]

**Pull request merged**